src/osd/ReplicatedPG.cc

```diff
@@ -5085,6 +5119,12 @@ int ReplicatedPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
       }
       break;
 
+    case CEPH_OSD_OP_WRITESAME:
+      ++ctx->num_write;
+      tracepoint(osd, do_osd_op_pre_writesame, soid.oid.name.c_str(), soid.snap.val, oi.size, op.writesame.offset, op.writesame.length, op.writesame.data_length);
+      result = do_writesame(ctx, osd_op);
```

Comment:
do_writesame() recurses with do_osd_ops(op = CEPH_OSD_OP_WRITE), so I don't think an extra write_update_size_and_usage() call is needed here.